kernel/kernel-api/src/main/java/io/delta/kernel/fs/FileReadRequest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.fs;
+// TODO here or in io.delta.kernel.client?
```

Comment:
only pending item

Reply:
ping TD and ask

Reply:
@tdas

Reply:
FileReadRequest is only used by Kernel internally to call TableClient.. right? Any users not implementing TableClient does not need this .. right? 


If yes to both, then it should be in client.

Reply:
Yes to both but can't the same thing be said for `FileStatus` which is in `io.delta.kernel.fs`?

Reply:
cc @vkorukanti 

is that true for FileStatus too? does Table/Snapshot/Scan API have no FileStatus in the parameter types or return types??

Reply:
`FileStatus` is used by the utility method that extracts the file status information from `Scan Row`. I think this should be moved to internal as well, given we moved the utility method to `InternalScanFileUtils`

Reply:
> FileReadRequest is only used by Kernel internally to call TableClient.. right? Any users not implementing TableClient does not need this .. right?
> 
> If yes to both, then it should be in client.

If we bo by this logic, then all expressions should be under `client`.

Reply:
`FileStatus` is in the `listFrom` return type though?

Reply:
Oh, got it. Then I guess we can move it under client.

Reply:
It gets bit messy under `client` package. Also we have expressions which are in separate package.